src/wp-includes/class-wp-theme-json.php

```diff
@@ -1934,6 +1934,15 @@ protected static function compute_style_properties( $styles, $settings = array()
 			return $declarations;
 		}
 
+		// The global settings can include dynamic data related to typography. We need evaluate it so that the cache is invalidated when it changes.
+		$args      = array( func_get_args(), wp_get_global_settings() ); // phpcs:ignore PHPCompatibility.FunctionUse.ArgumentFunctionsReportCurrentValue.Changed
```

Comment:
The way the theme.json code is organized is as follows:

1. `WP_Theme_JSON`: private class that deals with a `theme.json` structure. It doesn't depend on anything else.
2. `WP_Theme_JSON_Resolver`: private class that pulls data (theme.json files, theme.json from database) and merges it together appropriately (`get_merged_data`). It uses the `WP_Theme_JSON` class.
3. Public API: functions such as `wp_get_global_settings`, `wp_get_global_styles`, etc. They use and depend on the two previous private classes.

By using a public function such as `wp_get_global_settings` in the `WP_Theme_JSON` class, this code is creating a circular dependency. This code is saying: when processing _any_ theme.json structure, read all (core, blocks, theme, user) first.

Reply:
Thanks for the feedback @oandregal. This makes perfect sense. I think we would either need to add the caching to the public function that relies on `WP_Theme_JSON::compute_style_properties` or find a different approach to invalidation within `WP_Theme_JSON`.